### ISSUE TITLE
Copy/Paste Broken on non SSL Sites: `navigator.clipboard` is undefined

### ISSUE THREAD

**Issue by Alice-Deane** [pglombardo/PasswordPusher#734]
Hello,

Great work ! I love your docker image. 
Unfortunately copy/paste buttons won't work for me. I tried on firefox and chrome.

Is it just me ?
Any idea ?

Thanks,
Best regards.

**Comment by pglombardo** [comment]
Thanks @Alice-Deane - apologies for the issue though.  That is likely a javascript issue.

Could you tell me the following:

1. What docker container you are using (and tag)?
2. Could you open the javascript console and post any errors/warnings here?
3. Did it work previously or is this your first deploy?

**Comment by Alice-Deane** [comment]
Yes I looked up at the javascript and couldn't find any error. 
I am using the latest version of your docker image for my container and this is my first deploy.
I think it might be because I haven't configured https yet, still using http... I'll try to set it up !

**Comment by pglombardo** [comment]
> I am using the latest version of your docker image

I assume you mean the `latest` tag...

> Unfortunately copy/paste buttons won't work for me. I tried on firefox and chrome.

When you say "won't work" do you mean nothing gets copied to the clipboard?  Or something else?

Does the password generator work?  As that uses Javascript as well...  it might give an indication.

> I think it might be because I haven't configured https yet, still using http... I'll try to set it up !

I don't think that should make a difference since my local development environment doesn't use https...  but maybe...

**Comment by Alice-Deane** [comment]
Yes I do mean the `latest` tag.

> When you say "won't work" do you mean nothing gets copied to the clipboard? Or something else?

When I click on the different button to copy (for example the password or the link) and then I do paste the result it appear that nothing is being paste (so nothing has been copied). 

So I manually have to select the text, right click and then copy/paste the passwords...
I can't setup the https environment now (I don't know when I will be able to do it).  

> Does the password generator work? As that uses Javascript as well... it might give an indication.

Everything else is running perfectly, the password generator does work.

**Comment by Alice-Deane** [comment]
Hello, here is a quick recap of the issue : 
![image](https://user-images.githubusercontent.com/104282193/214516548-48049412-738a-4959-902d-a9cd22a61b57.png)
When I click on "Push it!" I land on this page. As you can see the button is already clicked but won't paste anything. I'm using firefox. 

![image](https://user-images.githubusercontent.com/104282193/214516838-468eff70-134c-4d59-8149-239398ad7223.png)
 Now if I go on the link (manually), and I click on "Copier dans le presse-papier" same issue, nothing happen. 

Here is the js alert I could see : 
![image](https://user-images.githubusercontent.com/104282193/214518171-245f7faf-1efe-4acd-9162-e8ad178939f7.png)

Thank you for your help !

**Comment by Alice-Deane** [comment]
I searched a little and it looks like it is due to the origin, it has to come from https or local and I'm in none of the cases. 
Apologies for the inconvenience. :)

**Comment by pglombardo** [comment]
I'm not sure if I understand what the issue was.  This was a CORS issue because of how you were hosting the app?

Let me know - I'd like to have the info in case someone else hits this issue (or maybe I can improve the code).

**Comment by Alice-Deane** [comment]
I'm not sure what a CORS issue is... But here are the results of my research : 

https://github.com/amatsuda/heavens_door/issues/15

Hope this will help you to understand

**Comment by pglombardo** [comment]
Thank you @Alice-Deane - that's a big help.  

I'll add a javascript fallback to fix this issue sometime soon.  I'm a bit tied up with some other things but I'll post back here once it's done/out.

**Comment by pglombardo** [comment]
Hi @Alice-Deane et. al. - just as a final summary on this issue for anyone else that comes across this issue:

The copy to clipboard button won't work on non-SSL connections.  This is a browser security feature and likely won't be fixed.

If hosted internally, use the [all-in-one deployment](https://docs.pwpush.com/docs/installation/#all-in-one-with-automatic-ssltls) with comes with SSL out of the box (or use your own SSL terminating proxy). 

tldr; pwpush needs to be access via SSL for the copy to clipboard button to work.

I'll close this issue out but if anything remains, please let me know!
